system/src/ble_listening_mode_handler.h

```diff
@@ -45,6 +63,12 @@ class BleListeningModeHandler {
     bool preAdvertising_;
     bool preConnected_;
     hal_ble_auto_adv_cfg_t preAutoAdv_;
+    bool userAdv_;
+
+    std::unique_ptr<uint8_t[]> ctrlReqAdvData_;
```

Comment:
We could potentially used a `Vector` instead, since it's exactly a pointer + size.

Reply:
Sounds good! 👍